### PR TITLE
Add GitHub Action to lint Python code

### PR DESCRIPTION
**Pull request by cclauss**
Output: https://github.com/cclauss/OpenStreetMap.by/actions

### CODE REVIEW

**Comment by Komzpa** [comment]
Hey @cclauss  - thanks for this PR. I've checked your link and it seems that github action fails on GeoIP install. Any chance you can fix it so I can merge it cleanly, or give some pointers? I don't know how to fix this in github actions layout.

**Comment by cclauss** [comment]
https://pypi.org/project/GeoIP has not been updated since 2014 and `pip install GeoIP` seems to broken these daze.  Perhaps `pip install geoip2` is the way to go... https://pypi.org/project/geoip2

**Comment by tbicr** [comment]
@cclauss , looks https://github.com/osm-by/OpenStreetMap.by/blob/master/index.py#L3 also require this changes in this case

**Comment by cclauss** [comment]
@tbicr That https://github.com/cclauss/OpenStreetMap.by/actions

**Comment by cclauss** [comment]
These tests pass and they are not about style violations.

https://flake8.pycqa.org/en/latest/user/error-codes.html

On the flake8 test selection, this PR does _not_ focus on "_style violations_" (the majority of flake8 error codes that [__psf/black__](https://github.com/psf/black) can autocorrect).  Instead these tests focus on runtime safety and correctness:
* E9 tests are about Python syntax errors usually raised because flake8 can not build an Abstract Syntax Tree (AST).  Often these issues are a sign of unused code or code that has not been ported to Python 3.  These would be compile-time errors in a compiled language but in a dynamic language like Python they result in the script halting/crashing on the user.
* F63 tests are usually about the confusion between identity and equality in Python.  Use ==/!= to compare str, bytes, and int literals is the classic case.  These are areas where __a == b__ is True but __a is b__ is False (or vice versa).  Python >= 3.8 will raise SyntaxWarnings on these instances.
* F7 tests logic errors and syntax errors in type hints
* F82 tests are almost always _undefined names_ which are usually a sign of a typo, missing imports, or code that has not been ported to Python 3.  These also would be compile-time errors in a compiled language but in Python a __NameError__ is raised which will halt/crash the script on the user.

**Comment by abitrolly** [comment]
@cclauss is it possible to run some test requests with `web.py`? To ensure that at least imports work as expected.

**Comment by cclauss** [comment]
Yes.  This PR is focused on linting but it does fire up pytest in allow failures mode.  This allows experimentation to do things like https://github.com/webpy/webpy/tree/master/tests in future test-related pull requests.

**Comment by abitrolly** [comment]
@cclauss why do you want to lint application that can not be even started locally? )

**Comment by cclauss** [comment]
Because syntax errors will crash the app no matter where it is run so it is better to catch them before our user do.

File: .github/workflows/lint_python.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - run: pip install codespell flake8 isort pytest
+      - if: matrix.python-version >= 3.6
+        run: |
+          pip install black
+          black --check . || true
+      - run: codespell --quiet-level=2 || true  # --ignore-words-list="" --skip=""
+      - run: flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+      - run: isort --recursive . || true
+      - run: pip install -r requirements.txt
+      - run: pytest . || true
```
Comment (tbicr) [inline .github/workflows/lint_python.yml:28]:
```suggestion
name: lint_python

on:
  pull_request:
  push:

jobs:
  lint_python:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@master
      - uses: actions/setup-python@master
        with:
          python-version: 3.8
      - run: sudo apt-get install libgeoip-dev -y
      - run: pip install flake8
      - run: flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
      - run: pip install -r requirements.txt
```

few comments:
1. for now python runs on py3.8 so I proposed to change it there to py3.8 too
2. current isort doesn't check anything, so it look should be `isort --check-only --diff .`, but it require additional changes, so I propose to add it in separeate PR
3. current black doesn't check anything, so as codebase py3.8+ only and also require additional changes I propose to remove matrix condition and add black in separate PR
4. current codespell doesn't check anything, but put potential issues in output, so I propose to add it in separate PR with rationalization description
5. current pytest doesn't check anything, this codebase don't have test, so I propose add tests and appropriate check in separate PR

**Review by tbicr — CHANGES_REQUESTED** [review]

**Review by tbicr — APPROVED** [review]
LGTM, thanks

**Comment by abitrolly** [comment]
Spring is coming. PRs are merging. Nice to see some collaboration going on. Keep it up guys. :)

**Pull request merged**